src/main/java/org/jenkinsci/plugins/jiraext/view/Transition.java

```diff
@@ -18,18 +18,18 @@
  **************************************************************************/
 package org.jenkinsci.plugins.jiraext.view;
 
-import com.google.inject.Inject;
 import hudson.Extension;
 import hudson.Launcher;
 import hudson.model.AbstractBuild;
 import hudson.model.BuildListener;
 import org.apache.commons.lang.StringUtils;
-import org.jenkinsci.plugins.jiraext.GuiceSingleton;
 import org.jenkinsci.plugins.jiraext.domain.JiraCommit;
-import org.jenkinsci.plugins.jiraext.svc.JiraClientSvc;
 import org.kohsuke.stapler.DataBoundConstructor;
 
+import java.util.Arrays;
+import java.util.Iterator;
 import java.util.List;
+import java.util.stream.Collectors;
 
 /**
  * Transition a JIRA issue
```

Comment:
This is a big enough change to warrant a new class. The previous behavior was determinate and easily communicated: try to call a transition on a ticket. If there's errors or it wasn't possible, log it and continue.

This new behavior is to try a transition on a ticket, and if that fails, call the next one until one works. Then go back and try all transitions.

What is the use-case here? Is it to get all tickets to a similar state (i.e closed)? IMO that is best solved with an update to Jira issue schema to allow such that as a single transition. Is there a reason that does not work, or is this for a different use-case entirely?